Error: Cannot find module "server.mjs"

### Environment

------------------------------
- Operating System: `Linux`
- Node Version:     `v14.19.1`
- Nuxt Version:     `2.16.0-27358576.777a4b7f`
- Package Manager:  `npm@8.6.0`
- Builder:          `webpack`
- User Config:      `rootDir`, `bridge`, `telemetry`, `nitro`, `head`, `css`, `plugins`, `components`, `buildModules`, `modules`, `compilerOptions`, `build`, `watchers`, `ssr`
- Runtime Modules:  `-`
- Build Modules:    `@nuxt/bridge@3.0.0-27495135.4d1a6ba`
------------------------------

### Reproduction

* Clone https://github.com/HendrikJan/nuxt-bridge-error-demo
* Run `npm install`
* Run `npx nuxi dev`
* Go to `http://localhost:3000/`
* See the error in your terminal and in your browser

### Describe the bug

Since a recent update I cannot get Nuxt to work anymore.
The application builds without errors, but when loading the page I get the 500 status with the error.

The error in the terminal is this:
```
✔ Builder initialized                                                                                                                                                                                  09:18:46
✔ Nuxt files generated                                                                                                                                                                                 09:18:46

✔ Client
  Compiled successfully in 3.12s

✔ Nitro built in 269 ms                                                                                                                                                                          nitro 09:18:50
ℹ Waiting for file changes                                                                                                                                                                             09:18:50
[nitro] [request error] Cannot find module '/home/haaiee/Projects/Docker/nuxt-bridge-error-demo/.nuxt/dist/server/server.mjs' imported from /home/haaiee/Projects/Docker/nuxt-bridge-error-demo/.nuxt/dev/index.mjs
  at new NodeError (internal/errors.js:322:7)  
  at finalizeResolution (internal/modules/esm/resolve.js:318:11)  
  at moduleResolve (internal/modules/esm/resolve.js:776:10)  
  at Loader.defaultResolve [as _resolve] (internal/modules/esm/resolve.js:887:11)  
  at Loader.resolve (internal/modules/esm/loader.js:89:40)  
  at Loader.getModuleJob (internal/modules/esm/loader.js:242:28)  
  at Loader.import (internal/modules/esm/loader.js:177:28)  
  at importModuleDynamically (internal/modules/esm/translators.js:114:35)  
  at exports.importModuleDynamicallyCallback (internal/process/esm_loader.js:30:14)  
  at ./.nuxt/dev/index.mjs:371:66
```

### Additional context

_No response_

### Logs

_No response_

--- a/src/nitro.ts
+++ b/src/nitro.ts
@@ -115,6 +115,9 @@ export async function setupNitroBridge () {
 
       // Nuxt aliases
       ...nuxt.options.alias
+    },
+    replace: {
+      'process.env.NUXT_NO_SSR': nuxt.options.ssr === false ? true : undefined
     }
   })
 

